docs/WritingTests.asciidoc

```diff
@@ -265,20 +265,25 @@ In short, writing multi-machine tests adds a few more layers of complexity:
 
 ==== Job dependencies
 
-There are 2 types of dependencies: +CHAINED+ and +PARALLEL+:
+There are 3 types of dependencies: +CHAINED+, +DIRECTLY_CHAINED+ and +PARALLEL+:
 
-* +CHAINED+ describes when one test case depends on another and both are run sequentially, i.e. KDE test suite is run after and only after Installation test suite
+* +CHAINED+ and +DIRECTLY_CHAINED+ describe when one test case depends on another and both are run sequentially, i.e. KDE test suite is run after and only after Installation test suite
 is successfully finished and cancelled if fail.
 
-To define +CHAINED+ dependency add variable +START_AFTER_TEST+ with the name(s) of test suite(s) after which selected test suite is supposed to run.
+The difference between +CHAINED+ and +DIRECTLY_CHAINED+ dependencies is that +DIRECTLY_CHAINED+ means the tests must run directly after another on the same worker slot.
+
+To define a +CHAINED+ dependency add the variable +START_AFTER_TEST+ with the name(s) of test suite(s) after which selected test suite is supposed to run.
 Use comma separated list for multiple test suite dependency. E.g. +START_AFTER_TEST="kde,dhcp-server"+
 
-* +PARALLEL+ describes MM test, test suites are scheduled to run at the same time and managed as a group. On top of that, +PARALLEL+ also describes
-test suites dependencies, where some test suites (children) run parallel with other test suites (parents) only when parents are running.
+To define a +DIRECTLY_CHAINED+ add the variable +START_DIRECTLY_AFTER_TEST+. It works in the same way as for +CHAINED+ dependencies. Make sure all tests have the same worker class.
```

Comment:
I think you are missing a word "dependency" after `+DIRECTLY_CHAINED+`. Also, what do you mean by "Make sure all tests have the same worker class."? That should not be anything special for directly chained. It would probably not make much sense to use `DIRECTLY_CHAINED` though unless you ensure to match even the same worker instance, which is not at all ensured by the worker class, isn't it?

Reply:
I mean that there must be a single worker which provides all the classes in the "chained cluster" at the same time. To simplify this, I've just written that users should use the same worker class for their directly chained jobs.

I don't think that's necessarily the case for other dependencies since the jobs can run on different workers.

Reply:
The wording in your response here seems clearer ie. there must be a single worker which provides all the classes.

What happens if that's not the case? Will the code handle this and create an error? That seems like something that should also be mentioned here.